src/cpp/dpm_sampler.cpp

```diff
@@ -44,15 +45,60 @@
 #include <nmmintrin.h>
 #endif
 
-#include <gsl/gsl_randist.h>
-#include <gsl/gsl_rng.h>
-#include <gsl/gsl_sf_exp.h>
-#include <gsl/gsl_sf_log.h>
-#include <gsl/gsl_sf_pow_int.h>
-
 #include "data_structures.hpp"
 #include "dpm_sampler.hpp"
 
+#include <random>
+#include <boost/random/beta_distribution.hpp>
+
+namespace {
+    std::mt19937 rg;
+
+    // Simple naive implementation of a discrete distribution, using cumulated weights.
+    // For generating a single number, that's enough.
+    //
+    // (as opposed to gsl's, STL's and boost's implementations that all allocate and compute extra look-up tables
+    // to further accelerate multiple futur calls to generator, see:
+    //  - https://www.gnu.org/software/gsl/manual/html_node/General-Discrete-Distributions.html
+    //  - https://stats.stackexchange.com/a/26868 )
+    template<typename UNS, typename CDF_type> inline UNS one_shot_discrete_cdf(CDF_type& cdf, const unsigned n, const double* wa) {
```

Comment:
remove `inline`, is pointless in 2020

Reply:
Get off my lawn! 
When I was your age, we needed to _inline_ bafefoot in the snow uphill!!! _Both ways!!!_
:rofl:

Reply:
Old fart's jokes aside:
Go actually test it in Godbolt, GCC actually still pays a bit attention to `inline` keywords (it doesn't blindly follow it like 90s compiler but still considers the suggestion) and in the present case, eventually takes the decision to not generate function instances.

Reply:
I don't care, `clang-tidy` recommends dropping it, and whether an external symbol is emitted is immaterial (`inline` functions can still emit external symbols if not inlined).